apps/cli/src/commands/boot-challenger.ts

```diff
@@ -114,26 +116,82 @@ const onAssertionConfirmedCb = async (nodeNum: any, commandInstance: Vorpal.Comm
+
+            try {
+                // Get the confirm data for all of the assertions
+                const [confirmData, confirmDataHash] = await getMultipleChallengeConfirmData(assertionIds);
+    
+                commandInstance.log(`[${new Date().toISOString()}] Submitting a batch challenge for assertion ids ${assertionIds}.`);
+    
+                // Submit a Batched Challenge TODO Confirm this is the correct way to submit a batch challenge
+                await submitAssertionToReferee(
+                    cachedSecretKey,
+                    nodeNum,
+                    lastSubmittedAssertionId,
+                    confirmDataHash,
+                    assertionNode.createdAtBlock,
+                    cachedSigner!.signer,
+                );
+    
+                // Update the last submitted assertion id and time
+                lastSubmittedAssertionId = nodeNum;
+                lastAssertionTime = Date.now();
+            } catch (error) {
+                if (error && (error as Error).message && (error as Error).message.includes('execution reverted: "9"')) {
+                    commandInstance.log(`[${new Date().toISOString()}] Could not submit challenge because it was already submitted`);
+                    lastAssertionTime = Date.now();
+                    return;
+                }
+                commandInstance.log(`[${new Date().toISOString()}] Submit Batch Assertion Error: ${(error as Error).message}`);
+                sendNotification(`Submit Batch Assertion Error: ${(error as Error).message}`, commandInstance);
+                throw error;
+                
+            }
+        }else{
+
+            // Submit for a single node event same as before
+            try {
+                await submitAssertionToReferee(
+                    cachedSecretKey,
+                    nodeNum,
+                    nodeNum - 1,
+                    assertionNode.confirmData,
+                    assertionNode.createdAtBlock,
+                    cachedSigner!.signer,
+                );
+                commandInstance.log(`[${new Date().toISOString()}] Submitted assertion: ${nodeNum}`);
+
+                // Update the last submitted assertion id and time
+                lastSubmittedAssertionId = nodeNum;
+                lastAssertionTime = Date.now();
+            } catch (error) {
+                if (error && (error as Error).message && (error as Error).message.includes('execution reverted: "9"')) {
+                    commandInstance.log(`[${new Date().toISOString()}] Could not submit challenge because it was already submitted`);
+                    lastAssertionTime = Date.now();
+                    return;
+                }
+                commandInstance.log(`[${new Date().toISOString()}] Submit Assertion Error: ${(error as Error).message}`);
+                sendNotification(`Submit Assertion Error: ${(error as Error).message}`, commandInstance);
+                throw error;
+            }
```

Comment:
This has a lot of duplicate code.
Can we not put most of that logic in the core function `submitAssertionToReferee` and let it handle hashing the confirmData if its more than one ?

Reply:
This has been refactored to simply set the confirm data in the if statement and then only a single reference is needed to submitAssertionToReferee.

I should have done this originally.